ice-adapter/src/main/java/com/faforever/iceadapter/ice/PeerIceModule.java

```diff
@@ -285,12 +300,14 @@ private void startIce() {
 
         // Wait for termination/completion of the agent
         long iceStartTime = System.currentTimeMillis();
-        while (agent.getState()
-                != IceProcessingState.COMPLETED) { // TODO include more?, maybe stop on COMPLETED, is that to early?
+        while (!Thread.currentThread().isInterrupted()
+                && agent.getState() != IceProcessingState.COMPLETED) { // TODO include more?, maybe stop on COMPLETED, is that to early?
             try {
                 Thread.sleep(20);
             } catch (InterruptedException e) {
                 log.error(getLogPrefix() + "Interrupted while waiting for ICE", e);
+                onConnectionLost();
```

Comment:
I added it on my own. I think if interrupted, then it is possible to terminate the connection.